### ISSUE TITLE
mason run --example --force doesn't force building examples

### ISSUE THREAD

**Issue by arezaii** [chapel-lang/chapel#18486]
### Summary of Problem
The mason run command can be given with the `--example` flag to either show all examples, run a specific example, or run all examples. The run command documents a `--build` flag which tells run to first build the examples. It also has a hidden `--force` flag which should tell mason to build the examples even if no source file changes are detected. The unexpected behavior here is that using the `--force` flag without the `--build` flag doesn't build the project, so `--force` is ignored if `--build` isn't also specified. 

I would expect that the `--force` flag should also trigger the build process, even if `--build` is not specified. 

### Steps to Reproduce

Start in a mason project with examples. A demo project is available at https://github.com/arezaii/argParser

list all examples:
```
 $ mason run --example
```

before running examples you need to build them:
```
$ mason build --example
```

run all the examples, attempting to force them to re-build:
```
$ mason run --example --force
```

The result is that the example just runs without building.

**Comment by dlongnecke-cray** [comment]
First off: If `--force` is a non-documented flag, I think we should feel at liberty to adjust its effects on other flags as we see fit.

In this case I actually think it's the `--build` flag that is strange. I would expect that `--example` builds the examples if they have not been built yet. And if a user throws `--build`, I would expect that to always build the examples (in effect, the same behavior as `--force`).

(Semantics of `--force`: Normally, when I see `--force`, the semantics tend to be "it doesn't matter if this thing might be unsafe, just do it". But here we're just telling `mason` to do an eager rebuild.)

Given the semantics of `--force` as I described above, it actually doesn't make sense in my mind for `--example --force` to rebuild the examples, because I just tried to force _running_ the examples (and of course, there's nothing to force there - running the examples always succeeds).

Anyhow in my mind the ideal behavior would be:

- First time use of `--example` builds the examples for you
- Use of `--example` will lazily rebuild any examples if their source files have changed

But that second bullet might be too aggressive for some. In which case you could use `--build` to force a rebuild.

**Comment by bradcray** [comment]
I tend to agree with David's response, where I'd probably have `run --force` have the behavior of emitting a warning saying "the `--force` flag has no effect here" rather than having it do more than run (i.e., build too) or just be silently ignored.
